airbyte-cdk/python/airbyte_cdk/sources/declarative/checks/check_stream.py

```diff
@@ -36,26 +36,10 @@ def check_connection(self, source: Source, logger: logging.Logger, config: Mappi
             if stream_name in stream_name_to_stream.keys():
                 stream = stream_name_to_stream[stream_name]
                 try:
-                    # Some streams need a stream slice to read records (eg if they have a SubstreamSlicer)
-                    stream_slice = self._get_stream_slice(stream)
-                    records = stream.read_records(sync_mode=SyncMode.full_refresh, stream_slice=stream_slice)
-                    next(records)
+                    stream_helper = StreamHelper()
+                    stream_helper.get_first_record(stream)
```

Comment:
Yes that actually makes a lot of sense! The only issue I see is that they don't _quite_ have the same functionality -- the `HTTPAvailabilityStrategy` currently handles 403 errors, but excepts other HTTPErrors and returns that the stream is still available. The check_stream behavior at the moment would still raise other HTTPErrors.  

@alafanechere I think this brings up a good point - right now the default behavior is other non-403 HTTPErrors will still cause issues and fail the whole sync - but while syncing (and potentially after having synced other streams), not during this check. Maybe we should make the default behavior do something like (pseudocode)
```
except HTTPError as http_error:
  if 403: 
    return False, "got 403"
  else:
    raise http_error
```

That would make us be able to do what @girarda suggested here, and make unexpected HTTPErrors fail the sync before it starts rather than after some streams have already synced.